### PR TITLE
Feature/ms 1709/rex exploitation

### PR DESCRIPTION
**Pull request by thelightcosine**
This PR moves the rex/exploitation directory, and its associated data files into the new rex-exploitation gem. This includes SEH, Egghunter, Omleete Hunter, JS detection and obfuscations, and cmdstagers
## Verification

List the steps needed to make sure this thing works
- [x] **Verify** all tests pass

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -249,14 +250,20 @@ GEM
...
+      rex-text
     rex-java (0.1.2)
     rex-mime (0.1.1)
       rex-text
     rex-nop (0.1.0)
       rex-arch
     rex-ole (0.1.2)
       rex-text
-    rex-powershell (0.1.64)
+    rex-powershell (0.1.65)
```
Comment (bcook-r7) [inline Gemfile.lock]: I don't think you want to bump this yet without the other PR landed first.

**Comment by dmaloney-r7** [comment]
the powershell pr is now landed so that particular concern should not be an issue now

**Comment by dmaloney-r7** [comment]
rebased off master to pull latest

**Comment by busterb** [comment]
This doesn't look like it was ready before it was merged, see #7415, reverting.

**Pull request merged**